lib/matplotlib/cbook/__init__.py

```diff
@@ -365,6 +389,13 @@ def process(self, s, *args, **kwargs):
                     proxy(*args, **kwargs)
                 except ReferenceError:
                     self._remove_proxy(proxy)
+                # this does not capture KeyboardInterrupt, SystemExit,
+                # and GeneratorExit
+                except Exception as e:
```

Comment:
I think this is good enough for now, we are not doing the right thing in the case where the function we are proxying does not properly handle it's own weak-refs going away.